js/ai/src/prompt.ts

```diff
@@ -89,9 +83,9 @@ export interface ExecutablePrompt<
    * @param opt Options for the prompt template, including user input variables and custom model configuration options.
    * @returns the model response as a promise of `GenerateStreamResponse`.
    */
-  <Out extends O>(
-    input?: I,
-    opts?: PromptGenerateOptions<I, CustomOptions>
+  <In extends I, Out extends O, Opts extends CustomOptions>(
```

Comment:
I understand why we have `Out` to make the return type strongly typed but not why `In` or `Opts`. The latter two can be derived from the arguments, no?

Reply:
OK, I think doing it this way is sufficient... removed these overridable types for now... can reintroduce later if users ask for it.

```ts
await ai.prompt<typeof InputSchema, typeof OutputSchema>('recipe')(input)
```